test/python/test_ibmqjob_states.py

```diff
@@ -30,132 +31,139 @@ def setUp(self):
         job = self.run_with_api(QueuedAPI())
-        self.assertStatus(job, JobStatus.INITIALIZING)
+        self.assertEqual(job.status(), JobStatus.INITIALIZING)
 
         self.wait_for_initialization(job)
-        self.assertStatus(job, JobStatus.QUEUED)
+        self.assertEqual(job.status(), JobStatus.QUEUED)
 
         self._current_api.progress()
-        self.assertStatus(job, JobStatus.RUNNING)
+        self.assertEqual(job.status(), JobStatus.RUNNING)
 
         self._current_api.progress()
-        self.assertStatus(job, JobStatus.DONE)
+        self.assertEqual(job.status(), JobStatus.DONE)
 
     def test_status_flow_for_cancellable_job(self):
         job = self.run_with_api(CancellableAPI())
-        self.assertStatus(job, JobStatus.INITIALIZING)
+        self.assertEqual(job.status(), JobStatus.INITIALIZING)
 
         self.wait_for_initialization(job)
-        self.assertStatus(job, JobStatus.RUNNING)
+        self.assertEqual(job.status(), JobStatus.RUNNING)
 
         can_cancel = job.cancel()
         self.assertTrue(can_cancel)
 
         self._current_api.progress()
-        self.assertStatus(job, JobStatus.CANCELLED)
+        self.assertEqual(job.status(), JobStatus.CANCELLED)
 
     def test_status_flow_for_non_cancellable_job(self):
         job = self.run_with_api(NonCancellableAPI())
-        self.assertStatus(job, JobStatus.INITIALIZING)
+        self.assertEqual(job.status(), JobStatus.INITIALIZING)
 
         self.wait_for_initialization(job)
-        self.assertStatus(job, JobStatus.RUNNING)
+        self.assertEqual(job.status(), JobStatus.RUNNING)
 
         can_cancel = job.cancel()
         self.assertFalse(can_cancel)
 
         self._current_api.progress()
-        self.assertStatus(job, JobStatus.RUNNING)
+        self.assertEqual(job.status(), JobStatus.RUNNING)
 
     def test_status_flow_for_errored_cancellation(self):
+        """This is still on RUNNING state because the Job
+        couldn't be cancelled, and is not due to a server
+        internal error, it's just in a non-cancelable state,
+        so the server returns an ERROR response.
+        """
         job = self.run_with_api(ErroredCancellationAPI())
-        self.assertStatus(job, JobStatus.INITIALIZING)
+        self.assertEqual(job.status(), JobStatus.INITIALIZING)
 
         self.wait_for_initialization(job)
-        self.assertStatus(job, JobStatus.RUNNING)
-
+        self.assertEqual(job.status(), JobStatus.RUNNING)
         can_cancel = job.cancel()
         self.assertFalse(can_cancel)
-        self.assertIsInstance(job.exception, IBMQJobError)
-
-        self.assertStatus(job, JobStatus.RUNNING)
+        self.assertEqual(job.status(), JobStatus.RUNNING)
 
     def test_status_flow_for_invalid_job(self):
         job = self.run_with_api(UnableToInitializeAPI())
-        self.assertStatus(job, JobStatus.INITIALIZING)
+        # TODO This is very risky, if the status changes to ERROR too fast
```

Comment:
Can you file a but for trying to remove race conditions while testing job state flow?

Reply:
Done. Issue #775